### DISCUSSION TITLE
Avoiding redundant renders in large lists?

### DISCUSSION THREAD

**Discussion by Wahib** [pmndrs/zustand#554] [Q&A] [answered]
I store the count of N counters in Zustand (3.5.10). Clicking on a counter increments it's count but forces a re-render of all the items in the list. Why? I made sure to memoize the selector and use shallow equality comparison. Please help!

https://codesandbox.io/s/hopeful-herschel-sk8uk?file=/src/App.js

* React DevTools > Highlight updates when components render
![Screenshot from 2021-09-05 20-37-38](https://user-images.githubusercontent.com/2498506/132136291-fb1c2a14-2630-40fc-941a-50251fc1bf41.png)

**Comment by dai-shi** [comment, marked as answer]
Here you go: https://codesandbox.io/s/naughty-rumple-sxey9

**Reply by Wahib** [reply]
Thanks. Your proposed solution will work as long as the list of child elements (counters) is static. What if we can add/delete counters dynamically? I have modified your solution (https://codesandbox.io/s/flamboyant-pike-x012b?file=/src/App.js) to add a button for deleting counters. Now clicking the plus button avoids redundant renders but clicking the delete button again causes all the child elements to re-rendering.

**Reply by dai-shi** [reply]
https://codesandbox.io/s/staging-microservice-tmk8w
In such case, `React.memo` will help.

**Reply by Wahib** [reply]
That seems to do the trick. Thanks for all the help!
